src/resmom/requests.c

```diff
@@ -3423,7 +3423,7 @@ req_cpyfile(struct batch_request *preq)
 		}
 	}
 
-	if (dir == STAGE_DIR_OUT && direct_write_requested(pjob))
+	if ((pjob != NULL) && (dir == STAGE_DIR_OUT) && direct_write_requested(pjob))
```

Comment:
The inner parentheses aren't really necessary.

Reply:
In this case, I still prefer them to be included for  better readability.